### DISCUSSION TITLE
When debouncing an input, previous long running requests aren't cancelled at stack up

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [livewire/livewire#8341] [7. Bugs (archived)]
### Livewire version

3.4.10

### Laravel version

10.5.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Add an artificial delay to the `render` method so that when the input is typed in it takes some time.
2. Add `wire:model.live.debounce.100ms="search"` to an input
3. When typing really quickly, multiple HTTP requests stack up. Previous requests should cancel.

### Repository or Wirebox URL

none

### Additional information

My livewire component is my whole page for an SPA feel.

Here's my input:

```html
<x-text-input id="search" name="search" type="text" class="block w-full md:w-1/3 lg:w-1/4" placeholder="Search" wire:model.live.debounce.100ms="search" data-lpignore="true" />
```

And my component:

```php
<?php

namespace App\Livewire\Pages\Applications;

use Livewire\Attributes\Url;
use Illuminate\Support\Str;
use Livewire\WithPagination;
use Livewire\Component;
use App\Models\Application;

class ApplicationIndex extends Component
{
    use WithPagination;

    #[Url]
    public string $search = '';
    public string $sortField = 'created_at';
    public string $sortDirection = 'desc';
    public bool $showAdvancedFilters = false;
    public array $filters = [
        'product' => null
    ];

    public function sortBy(string $field)
    {
        $this->sortDirection = $this->sortField === $field
            ? $this->sortDirection = $this->sortDirection == 'asc' ? 'desc' : 'asc'
            : 'asc';

        $this->sortField = $field;
    }

    public function updatedSearch()
    {
        $this->resetPage();
    }

    public function updatedFilters()
    {
        $this->resetPage();
    }

    public function render()
    {
        $this->authorize('viewAny', Application::class);

        sleep(3); // <-- here lies the problem?? 

        $applications = Application::withCount('entries')->with([
            'entry'
        ])->when($this->filters['product'], fn ($query, $product) => $query->where('product_type', $product))
            ->searchExact(['id', 'hash', 'email', 'mobile', 'birthday', 'postcode'], $this->search)
            ->orderBy($this->sortField, $this->sortDirection)
            ->simplePaginate(10);

        return view('livewire.pages.applications.application-index', [
            'applications' => $applications
        ]);
    }
}
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by iBotPeaches** [comment]
We just hit this. The older requests come back faster than the newer ones and the table of results jump around. Sometimes the last call is an older one and all sort of things break. Surprised this wasn't supported as it seems to go hand in hand with debouncing.

I started messing around with this to learn the internals of Livewire - https://github.com/iBotPeaches/livewire/pull/1/changes - this works for me, but far from ready. Its probably not something can be automatic and probably needs to be opt-in, since some operations may depend on the older calls while others prefer the cancellation.

Also mentioned:
 - https://github.com/livewire/livewire/discussions/2480
 - https://github.com/livewire/livewire/discussions/9030

**Reply by iBotPeaches** [reply]
The cancellation method doesn't work well without a large rewrite of stuff. The network calls can be batched with other things, so at that point you can't really cancel the other ones without negatively affecting something. I guess the downside of batching network calls when some of those you intentionally want to toss.

So I started thinking of a different method where we just incrementally go up per component on a number, so if a value comes back out of order for any ".live" response. We know we can ignore it since another response is coming in.

Dispatched Copilot to test it out - promising so far - https://github.com/iBotPeaches/livewire/pull/3/changes
